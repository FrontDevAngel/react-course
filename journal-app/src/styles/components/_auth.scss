.auth {
    &__main {
        background-color: $light-color;
        height: 100vh;
    }

    &__container {
        display: flex;
        margin: 0 auto;
        max-width: 960px;
        padding-bottom: 80px;
        padding-top: 160px;
    }

    &__title {
        color: $primary-color;
    }

    &__col {
        &--intro {
            flex: 1 0 0px;
            padding-right: 48px;
        }

        &--form {
            background-color: #ffffff;
            box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1),
                0 8px 16px rgba(0, 0, 0, 0.1);
            border-radius: 10px;
            padding: 25px;
        }
    }
}

.form {
    &--auth {
        text-align: center;
    }

    &__input,
    &__submit {
        margin-bottom: 15px;
    }

    &__submit,
    &__link {
        letter-spacing: 1.2px;
    }

    &__input {
        border-radius: 5px;
        border: 1px $primary-color solid;
        display: block;
        font-size: 18px;
        padding: 12px 18px;
        width: 364px;
        transition: ease box-shadow 0.3s;

        &:focus {
            outline: none;
            box-shadow: 0 2px 4px rgba($primary-color, 0.4);
        }
    }

    &__submit {
        background-color: $primary-color;
        border: 0 transparent none;
        border-radius: 8px;
        color: #ffffff;
        cursor: pointer;
        display: block;
        font-size: 16px;
        font-weight: 600;
        padding: 18px 23px;
        text-align: center;
        text-transform: uppercase;
        transition: ease background-color 0.3s;
        width: 100%;

        &:hover {
            background-color: $secondary-color;
        }
    }

    &__link {
        font-size: 16px;
        color: $primary-color;
        transition: ease color 0.3s;
        &:hover {
            color: $secondary-color;
        }
    }
}
