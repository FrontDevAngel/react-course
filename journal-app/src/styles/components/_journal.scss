.journal {
    &__container {
        display: flex;
        flex-direction: row;
        height: 100vh;
    }

    &__main,
    &__header {
        padding: 25px;
        box-sizing: border-box;
    }

    &__main {
        align-items: flex-start;
        flex-grow: 1;
    }

    &__header {
        background-color: $secondary-color;
        color: $light-color;
        height: 100%;
        width: 350px;
    }
}

.toolbar {
    &__header {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        margin-bottom: 25px;
    }
}
