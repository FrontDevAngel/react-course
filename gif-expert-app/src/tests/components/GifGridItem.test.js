import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Test de GifGridItem.test.js', () => {
    const imageData = {
        title: 'I See You Soccer GIF by UEFA',
        url:
            'https://media1.giphy.com/media/Ve7wX45gaOFmw8eeEM/giphy.gif?cid=755024f7e5gkbi9myo8ucsrnlsszjpk7uzacr9t3kxh0xffg&rid=giphy.gif',
    };

    const wrapper = shallow(<GifGridItem {...imageData} />);

    test('Test del componente renderizado comparado con snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un párrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(imageData.title);
    });

    test('Debe de igualar el source y alternative name de la imagen con el objeto prueba', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(imageData.url);
        expect(img.prop('alt')).toBe(imageData.title);
    });

    test('Evaluar que el div contenga la clase de animación : animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        // Ambas analizan lo mismo pero con diferentes metodos
        expect(className.includes('animate__fadeIn')).toBe(true);
        expect(className).toContain('animate__fadeIn');
    });
});
