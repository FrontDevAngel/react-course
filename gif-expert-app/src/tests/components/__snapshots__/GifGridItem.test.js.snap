// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifGridItem.test.js Test del componente renderizado comparado con snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="I See You Soccer GIF by UEFA"
    src="https://media1.giphy.com/media/Ve7wX45gaOFmw8eeEM/giphy.gif?cid=755024f7e5gkbi9myo8ucsrnlsszjpk7uzacr9t3kxh0xffg&rid=giphy.gif"
  />
  <p>
    I See You Soccer GIF by UEFA
  </p>
</div>
`;
